src/bundle/Core/Command/ReindexCommand.php

```diff
@@ -454,6 +482,32 @@ public function getDeprecatedAliases(): array
     {
         return ['ezplatform:reindex'];
     }
+
+    /**
+     * @param ArrayIterator<int, Content> $contentListIterator
+     * @param int $iterationCount
+     *
+     * @return \Generator
+     */
+    private function fetchIterationFromContentListIterator(ArrayIterator $contentListIterator, int $iterationCount): Generator
+    {
+        do {
+            $contentIds = [];
+            for ($i = 0; $i < $iterationCount; ++$i) {
+                $content = $contentListIterator->current();
+                if ($content) {
+                    $contentIds[] = $content->id;
+                } elseif (empty($contentIds)) {
+                    return;
+                } else {
+                    break;
+                }
+                $contentListIterator->next();
+            }
+
+            yield $contentIds;
+        } while (!empty($content));
```

Comment:
Performing a `foreach` operation and breaking early if `$iterationCount` is reached should be both less error prone and more readable.

Additionally it will probably remove your reliance on `Iterator`-specific implementation.

```suggestion
        $i = 0;
        $contentIds = [];
        foreach ($contentListIterator as $content) {
            $contentIds[] = $content->id;

            if ($iterationCount <= $i) {
                break;
            }
            $i++;
        }

        yield $contentIds;
```

Reply:
This looks better, Thanks for the suggestion! :)